[Concurrent Segment Search] SimpleNestedIT tests using nested sort are flaky for concurrent segment search

**Describe the bug**
The following tests are flaky for concurrent segment search:
- testSortNestedWithNestedFilter
- testNestedSortWithMultiLevelFiltering
- testLeakingSortValues
- testNestedSortingWithNestedFilterAsFilter
- testSimpleNestedSortingWithNestedFilterMissing
- testSimpleNestedSorting

See: https://build.ci.opensearch.org/job/gradle-check/29891
```
java.lang.AssertionError: Count is 0 hits but 3 was expected.  Total shards: 2 Successful shards: 1 & 1 shard failures:
 shard [[5Kj_N13LQg2fToy_tNYNEg][test][0]], reason [RemoteTransportException[[node_s1][127.0.0.1:39609][indices:data/read/search[phase/query/id]]]; nested: QueryPhaseExecutionException[Query Failed [Failed to execute main query]]; nested: NotSerializableExceptionWrapper[no_such_element_exception: null]; ], cause [NotSerializableExceptionWrapper[no_such_element_exception: null]
	at java.util.LinkedList.removeFirst(LinkedList.java:281)
	at java.util.LinkedList.pop(LinkedList.java:812)
	at org.opensearch.index.query.support.NestedScope.previousLevel(NestedScope.java:69)
	at org.opensearch.search.sort.SortBuilder.resolveNestedQuery(SortBuilder.java:250)
	at org.opensearch.search.sort.SortBuilder.resolveNested(SortBuilder.java:200)
	at org.opensearch.search.sort.SortBuilder.resolveNested(SortBuilder.java:196)
	at org.opensearch.search.sort.FieldSortBuilder.nested(FieldSortBuilder.java:583)
	at org.opensearch.search.sort.FieldSortBuilder.build(FieldSortBuilder.java:409)
	at org.opensearch.search.sort.SortBuilder.buildSort(SortBuilder.java:166)
	at org.opensearch.search.sort.FieldSortBuilder.getMinMaxOrNullInternal(FieldSortBuilder.java:629)
	at org.opensearch.search.sort.FieldSortBuilder.getMinMaxOrNullForSegment(FieldSortBuilder.java:624)
	at org.opensearch.search.internal.ContextIndexSearcher.canMatchSearchAfter(ContextIndexSearcher.java:512)
	at org.opensearch.search.internal.ContextIndexSearcher.canMatch(ContextIndexSearcher.java:504)
	at org.opensearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:295)
	at org.opensearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:280)
	at org.apache.lucene.search.IndexSearcher.lambda$search$1(IndexSearcher.java:715)
	at java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at org.opensearch.threadpool.TaskAwareRunnable.doRun(TaskAwareRunnable.java:78)
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
	at org.opensearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:59)
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:908)
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.lang.Thread.run(Thread.java:1583)
```
All tests show the same shard failure

**These tests have all been muted for concurrent segment search case in https://github.com/opensearch-project/OpenSearch/pull/11181 so we do not expect to see any test failures in Gradle Check**

--- a/server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java
@@ -501,10 +501,6 @@ public class SimpleNestedIT extends ParameterizedOpenSearchIntegTestCase {
     }
 
     public void testSimpleNestedSorting() throws Exception {
-        assumeFalse(
-            "Concurrent search case muted pending fix: https://github.com/opensearch-project/OpenSearch/issues/11187",
-            internalCluster().clusterService().getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
-        );
         assertAcked(
             prepareCreate("test").setSettings(Settings.builder().put(indexSettings()).put("index.refresh_interval", -1))
                 .setMapping(
@@ -604,10 +600,6 @@ public class SimpleNestedIT extends ParameterizedOpenSearchIntegTestCase {
     }
 
     public void testSimpleNestedSortingWithNestedFilterMissing() throws Exception {
-        assumeFalse(
-            "Concurrent search case muted pending fix: https://github.com/opensearch-project/OpenSearch/issues/11187",
-            internalCluster().clusterService().getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
-        );
         assertAcked(
             prepareCreate("test").setSettings(Settings.builder().put(indexSettings()).put("index.refresh_interval", -1))
                 .setMapping(
@@ -740,10 +732,6 @@ public class SimpleNestedIT extends ParameterizedOpenSearchIntegTestCase {
     }
 
     public void testNestedSortWithMultiLevelFiltering() throws Exception {
-        assumeFalse(
-            "Concurrent search case muted pending fix: https://github.com/opensearch-project/OpenSearch/issues/11187",
-            internalCluster().clusterService().getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
-        );
         assertAcked(
             prepareCreate("test").setMapping(
                 "{\n"
@@ -986,10 +974,6 @@ public class SimpleNestedIT extends ParameterizedOpenSearchIntegTestCase {
 
     // https://github.com/elastic/elasticsearch/issues/31554
     public void testLeakingSortValues() throws Exception {
-        assumeFalse(
-            "Concurrent search case muted pending fix: https://github.com/opensearch-project/OpenSearch/issues/11187",
-            internalCluster().clusterService().getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
-        );
         assertAcked(
             prepareCreate("test").setSettings(Settings.builder().put("number_of_shards", 1))
                 .setMapping(
@@ -1079,10 +1063,6 @@ public class SimpleNestedIT extends ParameterizedOpenSearchIntegTestCase {
     }
 
     public void testSortNestedWithNestedFilter() throws Exception {
-        assumeFalse(
-            "Concurrent search case muted pending fix: https://github.com/opensearch-project/OpenSearch/issues/11187",
-            internalCluster().clusterService().getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
-        );
         assertAcked(
             prepareCreate("test").setMapping(
                 XContentFactory.jsonBuilder()
@@ -1481,10 +1461,6 @@ public class SimpleNestedIT extends ParameterizedOpenSearchIntegTestCase {
 
     // Issue #9305
     public void testNestedSortingWithNestedFilterAsFilter() throws Exception {
-        assumeFalse(
-            "Concurrent search case muted pending fix: https://github.com/opensearch-project/OpenSearch/issues/11187",
-            internalCluster().clusterService().getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
-        );
         assertAcked(
             prepareCreate("test").setMapping(
                 jsonBuilder().startObject()

--- a/server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java
@@ -1863,10 +1863,6 @@ public class FieldSortIT extends ParameterizedOpenSearchIntegTestCase {
      * Test case for issue 6150: https://github.com/elastic/elasticsearch/issues/6150
      */
     public void testNestedSort() throws IOException, InterruptedException, ExecutionException {
-        assumeFalse(
-            "Concurrent search case muted pending fix: https://github.com/opensearch-project/OpenSearch/issues/11258",
-            internalCluster().clusterService().getClusterSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING)
-        );
         assertAcked(
             prepareCreate("test").setMapping(
                 XContentFactory.jsonBuilder()

